prime-angular-frontend/src/app/lib/modules/ngx-busy/busy.service.ts

```diff
@@ -18,14 +18,19 @@ export class BusyService {
   /**
    * @description
    * @param message the message to be displayed when a busy has a busy thing (subscription, promise, array of either... etc)
-   * @param request the observable that will run while the busy message and spinner are shown
+   * @param request the function or observable that will run while the busy message and spinner are shown
    * @returns OperatorFunction
    */
-  public showMessagePipe<T>(message: string, request: Observable<any>): OperatorFunction<T, Observable<any>> {
+  public showMessagePipe<T, R extends Function | Observable<any>>(message: string, request: R): UnaryFunction<Observable<any>, Observable<any>> {
+    const operator = exhaustMap((value: T) => (typeof request === 'function')
```

Comment:
You're close.  You can simplify this down to:

```js
  public showMessagePipe<T>(
    message: string,
    observableOrFunction: Observable<T> | (() => Observable<T>)
  ): UnaryFunction<Observable<unknown>, Observable<T>> {
    const operator = isObservable(observableOrFunction)
      ? exhaustMap(() => observableOrFunction)
      : exhaustMap(observableOrFunction);

    return pipe(
      tap((_) => (this.message = message)),
      operator,
      tap((_) => (this.message = ''))
    );
  }
```

It has:
1) More descriptive parameter, 
2) Doesn't use `any`
3) Uses `unknown` for the incoming value into the UnaryFunction
4) Uses `T` on the outgoing value from the UnaryFunction since that's known based on the Observable
4) Don't need to invoke the method passed, and 
5) Makes it easy for the next person to unpackage when scanning then having the ternary is inside the `exhaustMap`